crates/iroha/tests/sorting.rs

```diff
@@ -15,11 +15,10 @@ use nonzero_ext::nonzero;
 use rand::{seq::SliceRandom, thread_rng};
 
 #[test]
-#[ignore]
 #[allow(clippy::cast_possible_truncation)]
 fn correct_pagination_assets_after_creating_new_one() {
     // FIXME transaction is rejected for more than a certain number of instructions
-    const N_ASSETS: usize = 12;
+    const N_ASSETS: usize = 9;
```

Comment:
I think this test has failed for some other reason than the `FIXME` item. I see the same result by iterating over `submit_blocking` and there are no transaction rejections due to too many instructions in `submit_all_blocking`. In my local 10 and 11 are the threshold of whether it passes or not

Reply:
what's the default fetch size, 10? this might be related to result batching

Reply:
The test fails due to the lexicographical sorting of strings ("1" -> "10" -> "11", not "1" -> "2" -> "3").
All of the sorting tests have this limit of 10 because they will also fail if the number is increased.
This test should verify that adding an asset in the middle should not break the ordering, which was not checked in the previous version.

Reply:
So the issue would be that metadata leaf values are implicitly transmuted from numbers to strings

Reply:
Yes, since the metadata value is a json string. I'm not sure if we have a functionality to specify a mapping/key function for sorting. Isn't the purpose of having json string to generalize metadata and defer custom processing to the user?

Reply:
> It looks to me a bit surprising because I expect even JSON distinguishes between numbers and strings. What do you think @mversic ?

oh, this is because implementations for `PartialOrd`/`Ord` of our `primitives::Json` are derived on a string. Now I think they should be implemented by converting to `serde_json::Value` first

Reply:
but this might be too costly to convert every time. Here are some previous efforts to deal with this:
1. https://github.com/hyperledger-iroha/iroha/pull/5084/files#diff-4eb2755243c2b704928486450d5f707afd34df929926dbded2a9d585cf1c9ac3
2. https://github.com/hyperledger-iroha/iroha/pull/5012/files#diff-7cd63a689655313c90d3f2b1c9f49a5a2c37d0875c9b8c44d6265bef8b1824b2

Reply:
Since `PartialOrd` and `Ord` cannot be naturally derived from `serde_json::Value`, we could:

- totally discard the concept of sorting metadata value
- originally define order between `Value` variants

Reply:
> Since PartialOrd and Ord cannot be naturally derived from serde_json::Value, we could:

good point. My previous comment should be disregarded. Now I think this is the way to go:

> totally discard the concept of sorting metadata value

Reply:
If we go this way, the problem then is currently we are sorting query results only by metadata

Reply:
Can't we just sort by metadata key?

Reply:
[this](https://github.com/hyperledger-iroha/iroha/issues/2553) is where it all started. But since then we moved to untyped metadata value. Maybe it would be good to provide a sorting function?

Reply:
Yes, there seems to be a trade-off between keeping metadata values untyped and serving query sorting on the peer side